bundles/org.openhab.core/src/main/java/org/openhab/core/internal/items/ExpireManager.java

```diff
@@ -0,0 +1,367 @@
+        String itemName = commandEvent.getItemName();
+        logger.trace("Received command '{}' for item {}", command, itemName);
+        ExpireConfig expireConfig = getExpireConfig(itemName);
+        if (expireConfig != null) {
+            Command expireCommand = expireConfig.expireCommand;
+            State expireState = expireConfig.expireState;
+
+            if ((expireCommand != null && expireCommand.equals(command))
+                    || (expireState != null && expireState.equals(command))) {
+                // New command is expired command or state -> no further action needed
+                itemExpireMap.remove(itemName); // remove expire trigger until next update or command
+                logger.debug("Item {} received command '{}'; stopping any future expiration.", itemName, command);
+            } else {
+                // New command is not the expired command or state, so add the trigger to the map
+                Duration duration = expireConfig.duration;
+                itemExpireMap.put(itemName, Instant.now().plus(duration));
+                logger.debug("Item {} will expire (with '{}' {}) in {} ms", itemName,
+                        expireCommand == null ? expireState : expireCommand,
+                        expireCommand == null ? "state" : "command", duration);
+            }
+        }
+    }
+
+    protected void receiveUpdate(ItemStateEvent stateEvent) {
+        State state = stateEvent.getItemState();
+        String itemName = stateEvent.getItemName();
+        logger.trace("Received update '{}' for item {}", state, itemName);
+        ExpireConfig expireConfig = getExpireConfig(itemName);
+        if (expireConfig != null) {
+            Command expireCommand = expireConfig.expireCommand;
+            State expireState = expireConfig.expireState;
+
+            if ((expireCommand != null && expireCommand.equals(state))
+                    || (expireState != null && expireState.equals(state))) {
+                // New state is expired command or state -> no further action needed
+                itemExpireMap.remove(itemName); // remove expire trigger until next update or command
+                logger.debug("Item {} received update '{}'; stopping any future expiration.", itemName, state);
+            } else {
+                // New state is not the expired command or state, so add the trigger to the map
+                Duration duration = expireConfig.duration;
+                itemExpireMap.put(itemName, Instant.now().plus(duration));
+                logger.debug("Item {} will expire (with '{}' {}) in {} ms", itemName,
+                        expireCommand == null ? expireState : expireCommand,
+                        expireCommand == null ? "state" : "command", duration);
+            }
+        }
+    }
+
+    private @Nullable ExpireConfig getExpireConfig(String itemName) {
+        // the value might be null, so we explicitly check if an entry for that key exists
```

Comment:
I do not understand this comment. Where do we add a `null` value to the `itemExpireConfig` map? I guess your idea was to prevent repeated interaction with the metadata registry but currently it looks like we are trying to get and parse the metadata all the time unless an item has the new namespace.

Reply:
Good catch, this was indeed the intention.
With https://github.com/openhab/openhab-core/pull/1803/commits/79c256be3779703d783cae5a6afe597dd24b1752 in place, it should now indeed work as planned.

Reply:
That is good. A follow-up question. Metadata can be added, edited or removed during runtime e.g. via console or REST API. Will it be taken I to account too?

Reply:
You're always posing the right question... Indeed it wasn't taken into account, but it now is with https://github.com/openhab/openhab-core/pull/1803/commits/d7d71fbc903b4570d85d986573ba89c79cc08b18.